CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -1012,6 +1022,17 @@ public function updateKenyanProvinces(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Add index to status_id field as per CRM-20313.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   */
+  public function addIndexStatusId(CRM_Queue_TaskContext $ctx) {
+    $tables = array('civicrm_activity' => array('status_id'));
+    CRM_Core_BAO_SchemaHandler::createIndexes($tables);
+    return TRUE;
+  }
```

Comment:
IMO this should be a generic function following the same pattern as `\CRM_Upgrade_Incremental_Base::addColumn()` and `\CRM_Upgrade_Incremental_Base::dropColumn`.